### PR TITLE
Fix implicit size_t -> int conversions

### PR DESCRIPTION
**Pull request by orgads**
MSVC2015 warns about these when building for x64.

### CODE REVIEW

**Comment by orgads** [comment]
ping

**Comment by dimztimz** [comment]
This is a low priority issue (not a bug, just warnings), and i don't have time to review it right now. Please understand.

File: src/hunspell/affentry.cxx [hunk 1]
```diff
@@ -217,7 +217,7 @@ struct hentry* PfxEntry::checkword(const char* word,
   // and if there are enough chars in root word and added back strip chars
   // to meet the number of characters conditions, then test it
 
-  int tmpl = len - appnd.size(); // length of tmpword
+  int tmpl = len - int(appnd.size()); // length of tmpword
```
Comment (phajdan) [inline src/hunspell/affentry.cxx]: This is IMO a quite confusing/ambiguous way to write a cast.

Did you consider C++ style casts like static_cast? See http://cppblogs.blogspot.com/2013/08/c-style-casts.html for some examples.

Comment (orgads) [inline src/hunspell/affentry.cxx]: This is C++ style, but much shorter. The style is like an explicit constructor call. I don't find any advantage in static_cast compared to this. That's just longer and harder to read.

Comment (phajdan) [inline src/hunspell/affentry.cxx]: Please see https://google.github.io/styleguide/cppguide.html#Casting for some thoughts on this. Reasoning in above guide is very convincing for me. I suggested this as something which would make me much more comfortable merging this change.

**Comment by dimztimz** [comment]
This "fix" makes explicit what was previously implicit. It should not have change in behavior.

The proper way of fixing this is to use `size_t` everywhere where indexes are involved.

Although that is tricky as it can break things, e.g. backward iteration on array where the index was signed type. Switching simply to unsigned will break.

Therefore, I'm closing this because it is not a proper fix, but only more text to make those casts explicit.

**Comment by dimztimz** [comment]
We will be more careful for future code.

**Pull request closed without merging**